src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -153,7 +153,33 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
                 _kvSelectors.Add(new KeyValueSelector
                 {
                     KeyFilter = keyFilter,
-                    LabelFilter = labelFilter
+                    LabelFilter = labelFilter,
+                    SnapshotName = string.Empty
+                });
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Specify a snapshot and include its captured key-values in the configuration provider.
+        /// <see cref="SelectSnapshot"/> can be called multiple times to include key-values from multiple snapshots.
+        /// </summary>
+        /// <param name="name">The name of the snapshot in Azure App Configuration.</param>
+        public AzureAppConfigurationOptions SelectSnapshot(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (!_kvSelectors.Any(s => s.SnapshotName.Equals(name)))
+            {
+                _kvSelectors.Add(new KeyValueSelector
+                {
+                    KeyFilter = KeyFilter.Any,
```

Comment:
This doesn't need to be set does it? Same with label filter.

Reply:
I was setting these to avoid a null reference error in `LoadSelectedKeyValues` when we iterate through the key-value selectors. The other option was to check for null in that method instead, but I wasn't sure if it would be better to just prevent errors by setting the selector filters or just always check for null when referencing them. I think I'll just add the null checks because it makes more sense conceptually to have specific filters being set.